### PR TITLE
deal with new ruff version and next try for tutorial-test.yml

### PR DESCRIPTION
**Pull request by sroet**
this is the next try at testing the tutorial, will now only trigger on pushes to main.
(will have to merged before it triggers)

might solve #148 

Also tries to fix the ruff issues with the added ipynb ([a new ruff version has released with ipynb linting enabled](https://github.com/astral-sh/ruff/releases/tag/0.6.0)) and updates the pre-commit hook

Also updates minimal coverage to 87%

### CODE REVIEW

File: docs/benchmarks/defocus_gradient_analysis.ipynb [hunk 1]
```diff
@@ -13,8 +13,8 @@
     "import numpy as np\n",
     "from scipy.spatial.distance import cdist\n",
     "from scipy.optimize import curve_fit\n",
-    "from scipy.stats import t\n",
-    "sns.set(style='ticks', context='notebook')"
+    "\n",
```
Comment (McHaillet) [inline docs/benchmarks/defocus_gradient_analysis.ipynb:16]: Do all the double quotes also need to be escaped?

Comment (McHaillet) [reply docs/benchmarks/defocus_gradient_analysis.ipynb:16]: Nvm, they are inside double quotes

Comment (sroet) [reply docs/benchmarks/defocus_gradient_analysis.ipynb:16]: yeah, the double quotes and \n is how an ipynb looks like outside of jupyter

**Review by McHaillet — APPROVED** [review]
LGTM and anyway need to try the workflow right

**Pull request merged**